compiler/src/iree/compiler/Dialect/Stream/Transforms/EncodeTensors.cpp

```diff
@@ -440,6 +417,9 @@ struct EncodeTensorSliceOp
     // Dense:
     auto sourceOffset = calculateElementByteOffset(
         op.getLoc(), sourceType, sourceDims, op.getStartIndices(), rewriter);
+    if (!sourceOffset) {
+      return op.emitOpError("failed to calculate element byte offset");
```

Comment:
same here with better error messages - here and at all other sites in the file

Reply:
Actually taking a closer look this won't trigger at all now--the offset calculation is based on `Value`s so we can only check at runtime with some runtime asserts. Do we have such facilities right now? I can use `scf.assert` if that's okay.